client/session.go

```diff
@@ -574,99 +651,68 @@ func (s *session) Write(namespace, id string, t time.Time, value float64, unit x
 		return timestampErr
 	}
 
-	s.RLock()
-	if s.state != stateOpen {
+	if s.RLock(); s.state != stateOpen {
 		s.RUnlock()
 		return errSessionStateNotOpen
 	}
 
-	majority = atomic.LoadInt32(&s.majority)
+	c := s.writeStatePool.Get().(*writeState)
```

Comment:
Can we call this something like `write`?  Reading code below was hard to know what `c` was.